### PR TITLE
fix: address naming mixup and also kill some dead code

### PR DESCRIPTION
**Pull request by fidgetingbits**
I noticed some of the code I had that was ported was dead. Also there was a mixup with insert mode being called replace, and also missing select mode (although not used atm, I don't think it hurts to have it).

### CODE REVIEW

File: core/modes/mode_select.talon [hunk 1]
```diff
@@ -0,0 +1,4 @@
+win.title: /VIM MODE:s/
+-
+
+tag(): user.vim_select_mode
```
Comment (saidelike) [inline core/modes/mode_select.talon:4]: this should be `vim_mode_select`

**Comment by saidelike** [comment]
In general I don't think we should have dead code like the "select" mode. instead we could have a github issue with the code to actually use that in the future. That's generally what we do for the cursorless mono repo and I actually really like that

File: core/modes/switch_modes.py [hunk 2]
```diff
@@ -9,10 +9,10 @@
 # TODO: rename these function vim_mode_set_* or vim_mode_* instead of vim_set_*?
 @mod.action_class
 class Actions:
-    def vim_set_normal(auto: bool = True):
```
Comment (saidelike) [inline core/modes/switch_modes.py:12]: I don't recall what `auto` was used for?

File: core/modes/mode_select.talon (hunk 1 above)
Comment (saidelike) [reply core/modes/mode_select.talon:4]: we agreed to remove the select mode at the end, since we don't see any use for now

**Comment by saidelike** [comment]
@fidgetingbits I ended up reviewing this and fixing what we discussed. This is ready to be reviewed/merged

**Review by fidgetingbits — COMMENTED** [review]
Approve your additional changes. But same for this one you'll have to merge it seems since it was my PR.

**Review by saidelike — APPROVED** [review]

**Pull request merged**